1-Draft/RFCNNNN-Multi-Line-Continuance-For-Commands.md

```diff
@@ -0,0 +1,124 @@
+
+Instead, why not allow users to wrap commands across multiple lines in a more intuitive way without having to deal with backticks on every line or splatting, like this:
+
+```PowerShell
+New-ADUser @
+    -Name 'Jack Robinson'
+    -GivenName 'Jack'
+    -Surname 'Robinson'
+    -SamAccountName 'J.Robinson'
+    -UserPrincipalName 'J.Robinson@enterprise.com'
+    -Path 'OU=Users,DC=enterprise,DC=com'
+    -AccountPassword (Read-Host -AsSecureString 'Input Password')
+    -Enabled $true
+
+Get-ChildItem @
+    $rootFolder
+    -File
+    -Filter '*.ps*1'
+
+```
+
+Of course, they could invoke external commands and pass through arguments this way as well:
+
+```PowerShell
+& "./plink.exe" @
+    --% $Hostname -l $Username -pw $Password $Command
+
+cacls @
+    c:\docs\work
+    /E /T /C /G
+    "FinanceUsers":F
+
+```
+
+In each of these examples, the parser starts parsing the command as a multi-line command when it encounters the `@` token at the end of the line, and in this mode command parsing stops once one of the following is found:
+
+* end of file
+* blank line
+* command-terminating token
+
+The pros/cons to this new syntax are as follows:
+
+**Pros:**
+
+* allows the scripter to wrap commands how they see fit, while still getting Intellisense and tab completion, without using backticks.
+* parameters and arguments are entered the exact same way they would be if they were all placed on one line.
+* ad hoc could support this syntax as well (PSReadline could wait for a double-enter when in multi-line command parsing mode)
+* no breaking changes (a standalone @ is currently an unrecognized token in PowerShell no matter where it is used).
+
+**Cons:**
+
+* if no command-terminating token is used to terminate the command, a blank line is required after the command as terminator (but, users can opt to use a semi-colon or some other command-terminating token if they don't want the blank line)
+
+## Motivation
+
+    As a script/module author,
+    I can wrap commands across multiple lines easily and intuitively without backticks or splatting
+    so that my scripts remain easy to write and maintain while still giving me the benefits of Intellisense and tab completion.
+
+## Specification
+
+* expand the command parser to accept multi-line command parameters/arguments after an at symbol (`@`) is encountered at the end of a line
+* terminate multi-line commands when the parser encounters either a blank line or a command-terminating token (a command-terminating token will be a pipe symbol, redirection operator, closing enclosure, or semi-colon when the command does not have the stop-parsing sigil in its arguments, or simply pipe symbol or redirection operator the command does have the stop-parsing sigil in its arguments)
```

Comment:
Does this mean the same thing?:

* terminate multi-line commands when the parser encounters two newlines (rather than one), or any other normal command-terminating token

Reply:
To be clear, this is just a question -- I want to keep the detail here. I mean, I think it's necessary to spell it out explicitly for the spec.

Reply:
Yeah, I adopted your simpler wording and moved the detail down a littler further in the spec.